### PR TITLE
fix: remove usage of global disposeAfter from withFeature

### PR DESCRIPTION
**Pull request by barak007**

### CODE REVIEW

File: packages/test-kit/src/with-feature.ts [hunk 1]
```diff
@@ -561,26 +559,22 @@ async function enableTracing({
...
-                fs,
-                name:
-                    process.env.TRACING && process.env.TRACING !== 'true'
-                        ? process.env.TRACING
-                        : name ?? (testName ? normalizeTestName(testName) : 'nameless-test'),
-            }),
-        });
-    });
     dispose(
         async () => {
```
Comment (matangeorgi) [inline packages/test-kit/src/with-feature.ts:563]: the `async` is needed here?

**Comment by matangeorgi** [comment]
good change, organized a lot of stuff.

File: packages/test-kit/src/with-feature.ts (hunk 1 above)
Comment (barak007) [reply packages/test-kit/src/with-feature.ts]: not needed

File: packages/test-kit/src/run-environment.ts [hunk 2]
```diff
@@ -1,25 +1,24 @@
...
     /**@deprecated use engine.shutdown */
     dispose: () => Promise<void>;
 }> {
     const runningFeature = await originalGetRunningFeature(options);
-    if (disposeAfterTestTimeout) {
-        disposeAfter(runningFeature.engine.shutdown, {
-            name: `engine shutdown for ${options.featureName}`,
-            timeout: disposeAfterTestTimeout,
+    if (autoDisposeTimeout) {
+        afterEach(`engine shutdown for ${options.featureName}`, function () {
```
Comment (barak007) [inline packages/test-kit/src/run-environment.ts:19]: protect `afterEach`

**Pull request merged**